packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp

```diff
@@ -208,53 +248,19 @@ void ReferenceHexMesh::buildMesh(
                     i + j * x_num_cells + k * x_num_cells * y_num_cells;
                 DTK_CHECK( element_id < total_cells );
 
-                // node 0
-                node_id =
-                    ( i ) + (j)*x_num_nodes + (k)*x_num_nodes * y_num_nodes;
-                DTK_CHECK( node_id < total_nodes );
-                entity_set->getEntity( node_id, 0, hex_nodes[0] );
-
-                // node 1
-                node_id =
-                    ( i + 1 ) + (j)*x_num_nodes + (k)*x_num_nodes * y_num_nodes;
-                DTK_CHECK( node_id < total_nodes );
-                entity_set->getEntity( node_id, 0, hex_nodes[1] );
-
-                // node 2
-                node_id = ( i + 1 ) + ( j + 1 ) * x_num_nodes +
-                          (k)*x_num_nodes * y_num_nodes;
-                DTK_CHECK( node_id < total_nodes );
-                entity_set->getEntity( node_id, 0, hex_nodes[2] );
-
-                // node 3
-                node_id = ( i ) + ( j + 1 ) * x_num_nodes +
-                          (k)*x_num_nodes * y_num_nodes;
-                DTK_CHECK( node_id < total_nodes );
-                entity_set->getEntity( node_id, 0, hex_nodes[3] );
-
-                // node 4
-                node_id = ( i ) + (j)*x_num_nodes +
-                          ( k + 1 ) * x_num_nodes * y_num_nodes;
-                DTK_CHECK( node_id < total_nodes );
-                entity_set->getEntity( node_id, 0, hex_nodes[4] );
-
-                // node 5
-                node_id = ( i + 1 ) + (j)*x_num_nodes +
-                          ( k + 1 ) * x_num_nodes * y_num_nodes;
-                DTK_CHECK( node_id < total_nodes );
-                entity_set->getEntity( node_id, 0, hex_nodes[5] );
-
-                // node 6
-                node_id = ( i + 1 ) + ( j + 1 ) * x_num_nodes +
-                          ( k + 1 ) * x_num_nodes * y_num_nodes;
-                DTK_CHECK( node_id < total_nodes );
-                entity_set->getEntity( node_id, 0, hex_nodes[6] );
-
-                // node 7
-                node_id = ( i ) + ( j + 1 ) * x_num_nodes +
-                          ( k + 1 ) * x_num_nodes * y_num_nodes;
-                DTK_CHECK( node_id < total_nodes );
-                entity_set->getEntity( node_id, 0, hex_nodes[7] );
+#define GET_NODE( local_id, i, j, k )                                          \
```

Comment:
Don't think a function is a good choice. But did not think about lambdas. Will try to implement those.